smack-experimental/src/main/java/org/jivesoftware/smackx/mam/provider/MamResultProvider.java

```diff
@@ -0,0 +1,62 @@
+/**
+ *
+ * Copyright 2016 Fernando Ramirez
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.mam.provider;
+
+import org.jivesoftware.smack.provider.ExtensionElementProvider;
+import org.jivesoftware.smackx.forward.packet.Forwarded;
+import org.jivesoftware.smackx.forward.provider.ForwardedProvider;
+import org.jivesoftware.smackx.mam.element.MamElements.MamResultExtension;
+import org.xmlpull.v1.XmlPullParser;
+
+/**
+ * MAM Result Provider class.
+ * 
+ * @see <a href="http://xmpp.org/extensions/xep-0313.html">XEP-0313: Message
+ *      Archive Management</a>
+ *
+ */
+public class MamResultProvider extends ExtensionElementProvider<MamResultExtension> {
+
+    @Override
+    public MamResultExtension parse(XmlPullParser parser, int initialDepth) throws Exception {
+        String queryId = null;
+        String id = null;
+        Forwarded forwarded = null;
+
+        if (parser.getName().equals(MamResultExtension.ELEMENT)) {
+            queryId = parser.getAttributeValue("", "queryid");
+            id = parser.getAttributeValue("", "id");
+
+            boolean done = false;
+            while (!done) {
```

Comment:
Please remove `done` boolean and replace the String comparison with an integer one.